x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer_utils/time_series_search_service.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { each, find, get, filter } from 'lodash';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import type { MlEntityField } from '@kbn/ml-anomaly-utils';
+import type { Job } from '../../../../common/types/anomaly_detection_jobs';
+import type { ModelPlotOutput } from '../../services/results_service/result_service_rx';
+import type { MlApiServices } from '../../services/ml_api_service';
+import type { MlResultsService } from '../../services/results_service';
+import { buildConfigFromDetector } from '../../util/chart_config_builder';
+import {
+  isModelPlotChartableForDetector,
+  isModelPlotEnabled,
+} from '../../../../common/util/job_utils';
+
+// Note: This is a duplicated of `timeseries_search_service.ts` updated to move away from dependency cache.
+// The original file will be removed once all references are replaced with references to this file.
```

Comment:
Suggest to replace comment with the following so we can search for the pattern `// TODO Consolidate`:

```
// TODO Consolidate with legacy code in
// `x-pack/plugins/ml/public/application/timeseriesexplorer/timeseries_search_service.ts`
```